LINK/usr/bin/vmdb-up

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env ruby
+
+require "yaml"
+require "sd_notify"
+
+$stdout.sync = true
+
+def wait_for_service(host, port)
+  loop do
+    `ncat #{host} #{port} </dev/null 2>/dev/null`
+    break if $?.success?
+
+    puts "#{host} #{port} - not accepting connections"
+    sleep(10)
+  end
+
+  puts "#{host} #{port} - accepting connections"
+end
+
+def wait_for_vmdb
+  loop do
+    db_state = `cd /var/www/miq/vmdb; bin/rails r "require './lib/tasks/evm_application'; puts EvmApplication.server_state"`
```

Comment:
NOTE this is incredibly slow, investigate replacing with a stripped down table check

Reply:
I've been looking at v2-key parsing, which I think you'll need first - it's a little complicated, but I think I'm close

Reply:
```sh
$ echo $V2_KEY_FILE
certs/v2_key

$ echo $EXAMPLE_PASSWORD
v2:{LBQQb2c7PeXTV8Bgr9Z9ww==}

$ echo $EXAMPLE_PASSWORD | cut -c 5- | sed s/.$// | openssl enc -d -$(grep :algorithm: $V2_KEY_FILE | cut -d ' ' -f 2) -base64 -nosalt -iv "$(grep :iv: $V2_KEY_FILE | cut -d ' ' -f 2 | base64 -d | xxd -p -c256)" -K "$(grep :key: $V2_KEY_FILE | cut -d ' ' -f 2 | base64 -d | xxd -p -c256)"; echo
p4ssw0rd
```

Reply:
Broken down (not in order):

Remove the `v2:{` prefix and `}` suffix

```
echo $EXAMPLE_PASSWORD | cut -c 5- | sed s/.$//
```

Extract the algorithm, key, and iv from the v2 key yaml file.
```
$(grep :algorithm: $V2_KEY_FILE | cut -d ' ' -f 2)
$(grep :key: $V2_KEY_FILE | cut -d ' ' -f 2)
$(grep :iv: $V2_KEY_FILE | cut -d ' ' -f 2)
```

Convert the base64 encoded key or iv into the hexdigit form needed by openssl enc:

```
| base64 -d | xxd -p -c256
```

Run openssl enc -d:

```
openssl enc
-d          # decrypt
-$algorithm # with the specified algorithm (should be aes-256-cbc; from the v2 key)
-base64     # Base 64 decode incoming password
-iv "$iv"   # The iv in hexdigit format (from the v2 key)
-K "$key"   # The key in hexdigit format (from the v2 key)

Reply:
Okay I've replaced `bin/rails r "require './lib/tasks/evm_application'; puts EvmApplication.server_state"` with a simple table check

Reply:
> I've been looking at v2-key parsing, which I think you'll need first - it's a little complicated, but I think I'm close

haha, understatement of the year.  🏆